torch/testing/_internal/autograd_function_db.py

```diff
@@ -0,0 +1,250 @@
+import torch
+from functools import partial
+from torch.testing import make_tensor
+from torch.testing._internal.opinfo.core import (
+    OpInfo,
+    SampleInput,
+)
+from torch.testing._internal.common_dtype import all_types_and
+import numpy as np
+
+# Note: [autograd.Function db]
+#
+# This is a collection of autograd.Function test cases written as OpInfos
+# so they can easily be consumed by OpInfo-based tests to check if a subsystem
+# supports autograd.Function.
```

Comment:
cc @mruberry @ngimel -- In this PR, we've added some autograd.Function examples as OpInfos. I have not made this available by default in the `op_db` that all tests use, but send them through functorch tests because that's the easiest way to test the autograd.Function <-> functorch interaction. Should we make them available by default?

I haven't made them available by default because autograd.Function aren't really "operators", though I can see this being useful beyond just functorch testing.